### PR TITLE
Don't display security banner for unknown RecoveryState

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

- Prevents the 'set up recovery' banner from being displayed in the room list when the recovery state is unknown.
- Also prevents the UI for setting up the recovery from appearing in the chat backup options when the recovery state is unknown.
- Added some more screenshots to verify the different states.

## Motivation and context

@bnjbvr reported this bug found when the device couldn't correctly connect to the HS.

## Screenshots / GIFs

In the PR.

## Tests

The bug report says the device couldn't reach the server but the device had internet connectivity, so maybe block your HS address in your network temporarily or enable a proxy that doesn't connect to the HS.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/root/SecureBackupRootStateProvider.kt [hunk 1]
```diff
@@ -20,6 +20,13 @@ open class SecureBackupRootStateProvider : PreviewParameterProvider<SecureBackup
             aSecureBackupRootState(backupState = BackupState.UNKNOWN, doesBackupExistOnServer = AsyncData.Success(true)),
             aSecureBackupRootState(backupState = BackupState.UNKNOWN, doesBackupExistOnServer = AsyncData.Success(false)),
             aSecureBackupRootState(backupState = BackupState.UNKNOWN, doesBackupExistOnServer = AsyncData.Failure(Exception("An error"))),
+            aSecureBackupRootState(backupState = BackupState.WAITING_FOR_SYNC),
+            aSecureBackupRootState(backupState = BackupState.CREATING),
+            aSecureBackupRootState(backupState = BackupState.RESUMING),
+            aSecureBackupRootState(backupState = BackupState.RESUMING),
```
Comment (bmarty) [inline features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/root/SecureBackupRootStateProvider.kt:26]: There is a duplicated case here.

File: features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/root/SecureBackupRootStateProvider.kt [hunk 2]
```diff
@@ -20,6 +20,13 @@ open class SecureBackupRootStateProvider : PreviewParameterProvider<SecureBackup
...
+            aSecureBackupRootState(backupState = BackupState.DOWNLOADING),
+            aSecureBackupRootState(backupState = BackupState.DISABLING),
+            aSecureBackupRootState(backupState = BackupState.ENABLED),
```
Comment (bmarty) [inline features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/root/SecureBackupRootStateProvider.kt:29]: here too.

**Review by bmarty — COMMENTED** [review]
Make sense, thanks. A bit too many preview states added :)

File: features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/root/SecureBackupRootStateProvider.kt (hunk 1 above)
Comment (jmartinesp) [reply features/securebackup/impl/src/main/kotlin/io/element/android/features/securebackup/impl/root/SecureBackupRootStateProvider.kt:26]: Ooops, too many `Cmd+D`. Good catch!

**Review by bmarty — APPROVED** [review]

**Review by bmarty — APPROVED** [review]

**Pull request merged**